diktat-rules/src/test/kotlin/org/cqfn/diktat/ruleset/chapter3/files/NewlinesRuleWarnTest.kt

```diff
@@ -122,6 +122,42 @@ class NewlinesRuleWarnTest {
         )
     }
 
+    @Test
+    fun `line breaking after infix functions - positive example`() {
+        lintMethod(NewlinesRule(),
+                """
+                    |fun foo() {
+                    |    true xor
+                    |        false
+                    |        
+                    |    true
+                    |        .xor(false)
+                    |    
+                    |    (true xor
+                    |        false)
+                    |}
+                """.trimMargin()
+        )
+    }
+
+    @Test
+    fun `line breaking after infix functions`() {
+        lintMethod(NewlinesRule(),
+                """
+                    |fun foo() {
+                    |    (true 
```

Comment:
I would also have added a test with multiple infix functions like:
(a xor b xor c) - it will have a little bit different tree